cannon/mipsevm/tests/helpers.go

```diff
@@ -50,31 +50,50 @@ func multiThreadElfVmFactory(t require.TestingT, elfFile string, po mipsevm.Prei
 	return fpvm
 }
 
+type ThreadProofEncoder func(po mipsevm.PreimageOracle, stdOut, stdErr io.Writer, log log.Logger, opts ...testutil.StateOption) []byte
+
+func singalThreadProofEncoder(po mipsevm.PreimageOracle, stdOut, stdErr io.Writer, log log.Logger, opts ...testutil.StateOption) []byte {
+	return nil
+}
+
+func multiThreadProofEncoder(po mipsevm.PreimageOracle, stdOut, stdErr io.Writer, log log.Logger, opts ...testutil.StateOption) []byte {
+	state := multithreaded.CreateEmptyState()
+	mutator := mttestutil.NewStateMutatorMultiThreaded(state)
+	for _, opt := range opts {
+		opt(mutator)
+	}
+	proofData := state.EncodeThreadProof()
+	return proofData
+}
+
 type VersionedVMTestCase struct {
-	Name         string
-	Contracts    *testutil.ContractMetadata
-	StateHashFn  mipsevm.HashFn
-	VMFactory    VMFactory
-	ElfVMFactory ElfVMFactory
+	Name               string
+	Contracts          *testutil.ContractMetadata
+	StateHashFn        mipsevm.HashFn
+	VMFactory          VMFactory
+	ElfVMFactory       ElfVMFactory
+	ThreadProofEncoder ThreadProofEncoder
 }
 
 func GetSingleThreadedTestCase(t require.TestingT) VersionedVMTestCase {
 	return VersionedVMTestCase{
-		Name:         "single-threaded",
-		Contracts:    testutil.TestContractsSetup(t, testutil.MipsSingleThreaded),
-		StateHashFn:  singlethreaded.GetStateHashFn(),
-		VMFactory:    singleThreadedVmFactory,
-		ElfVMFactory: singleThreadElfVmFactory,
+		Name:               "single-threaded",
+		Contracts:          testutil.TestContractsSetup(t, testutil.MipsSingleThreaded),
+		StateHashFn:        singlethreaded.GetStateHashFn(),
+		VMFactory:          singleThreadedVmFactory,
+		ElfVMFactory:       singleThreadElfVmFactory,
+		ThreadProofEncoder: singalThreadProofEncoder,
```

Comment:
`ThreadProofEncoder` shouldn't be stored in `VersionedVMTestCase` as it's specific to mt-cannon. Simply make `multiThreadProofEncoder` package-visible so the multithreaded tests can call it directly.